pepdbagent/search.py

```diff
@@ -0,0 +1,240 @@
+import psycopg2
+from .models import (
+    ProjectSearchModel,
+    NamespaceSearchModel,
+    ProjectSearchResultModel,
+    NamespaceSearchResultModel,
+)
+import logging
+from .const import *
+from .base import BaseConnection
+from typing import List
+
+_LOGGER = logging.getLogger("pepdbagent")
+
+
+class Search(BaseConnection):
+    def __init__(self, db_conn: psycopg2):
+        """
+        Search class that holds 2 search methods: namespace and project search.
+        :param db_conn: already existing connection to the db
+        """
+        super().__init__(db_conn=db_conn)
+
+    def namespace(
+        self,
+        search_str: str = None,
+        admin_list: List[str] = None,
+        offset: int = DEFAULT_OFFSET,
+        limit: int = DEFAULT_LIMIT,
+    ) -> NamespaceSearchModel:
+        """
+        Search available namespaces in the database
+        :param search_str: search string
+        :param admin_list: list of namespaces where user is admin
+        :param offset: offset of the search
+        :param limit: limit of the search
+        :return: Search result:
+            {
+                total number of results
+                search limit
+                search offset
+                search results
+            }
+        """
+        if admin_list:
+            admin_list = tuple(admin_list)
+        else:
+            admin_list = ("",)
+        return NamespaceSearchModel(
+            number_of_results=self._get_count_of_found_namespaces(
+                search_str=search_str, admin_nsp=admin_list
+            ),
+            limit=limit,
+            offset=offset,
+            results=self._find_namespaces(
+                search_str=search_str, admin_nsp=admin_list, offset=offset, limit=limit
+            ),
+        )
+
+    def project(
```

Comment:
It's not clear why there's: `con.get_project`, and `con.get_raw_project`, and `con.search.project`.